app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java

```diff
@@ -0,0 +1,76 @@
+package com.iitp.njack.iitp_connect.core.youtube.video;
+
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.databinding.ObservableInt;
+import android.support.annotation.VisibleForTesting;
+import android.view.View;
+
+import com.iitp.njack.iitp_connect.data.youtube.YoutubeVideo;
+import com.iitp.njack.iitp_connect.utils.RateLimiter;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class VideoViewModel extends ViewModel {
+    @VisibleForTesting
+    static final String VIDEO_VIEW_MODEL = "videoViewModel";
+
+    private final RateLimiter<String> repoRateListLimit;
+    public ObservableInt loading;
+    public ObservableInt showEmpty;
```

Comment:
Any way we could use `SingleEventLiveData`?

Reply:
@sheetal-g9 ?

Reply:
I am using this just to observe the visibility of a layout which is of primitive type int.
ObservableInt (which is a subclass of databinding.BaseObservable )would be more suitable here.

Reply:
The use of `showEmpty` is just to signal the Activity that you should show an empty view layout. That's the exact purpose of a `SingleEventLiveData`. It is used to send a one-time signal to the view about something.